pykickstart/commands/autopart.py

```diff
@@ -654,3 +654,38 @@ def _getParser(self):
 class RHEL9_AutoPart(RHEL8_AutoPart):
     pass
 
+
+class F38_AutoPart(F29_AutoPart):
+    def __init__(self, writePriority=100, *args, **kwargs):
+        F29_AutoPart.__init__(self, writePriority=writePriority, *args, **kwargs)
+        self.hibernation = kwargs.get("hibernation", False)
+        self.swap = kwargs.get("swap", False)
+
+    def __str__(self):
+        retval = F29_AutoPart.__str__(self)
+        if not self.autopart:
+            return retval
+
+        if self.hibernation:
+            # remove any trailing newline
+            retval = retval.strip()
+            retval += " --hibernation"
+            retval += "\n"
+
+        return retval
+
+    def _getParser(self):
+        op = F29_AutoPart._getParser(self)
+        op.add_argument("--hibernation", action="store_true", default=False,
+                        version=F38, help="""
+                        Create larger swap for hibernation.""")
```

Comment:
Maybe it could say something like: `Create a swap partition with an automatically determined size that's big enough for hibernation.` You might also mention, that the `--hibernation` option shouldn't be used together with `--noswap`.

Reply:
>  You might also mention, that the --hibernation option shouldn't be used together with --noswap.
Isn't this obvious to the average Kickstart user? Also If I include this warning in `--hibernation` I should probably add it to `--noswap` as well.

Reply:
Actually, yes, both places should have the note.